docs/RFD-JobWorkerService.md

```diff
@@ -0,0 +1,366 @@
+
+* `Start` New Job
+  * verify user credentials via mTLS Authorization
+  * create new instance of the `Job` and assign new UUID 
+  * store `job` in sync.Map using UUID as key
+  * start command execution
+  * return UUID to the client
+* `Status` of the Job
+  * return current status of the `Job`
+* `Stream` latest Job output
+  * lookup `Job` by provided UUID
+  * request instance of `OutputReader` using library
+  * send steam output to the client/cli
+* `Stop` Job
+  * lookup `Job` by provided UUID
+  * call `Stop` function if `Job` was found, otherwise return error
+  
+  > **Not any job and it output removed from then server**. So, depend on the load and size of total output it would come to OOM situation sooner or later and [panic](https://go.dev/blog/defer-panic-and-recover).
+  > Probably we need to consider some API or process to clean completed jobs based on size, date of creation, frequency of usage or etc.
+  
+
+#### Security
+
+The client and API communicate via mTLS using [TLS 1.3 ](https://datatracker.ietf.org/doc/html/rfc8446) as the minimum version. The following cipher suites are supported:
+
+* TLS_AES_256_GCM_SHA384
+* TLS_CHACHA20_POLY1305_SHA256
+* TLS_AES_128_GCM_SHA256
+
+> Note: only 3 cipher suites enabled openssl by default: https://wiki.openssl.org/index.php/TLS1.3, and following 2 
+> below also can be used, but need to explicitly enabled: 
+>* TLS_AES_128_CCM_8_SHA256
+>* TLS_AES_128_CCM_SHA256
+
+> Since there are no requirements for older versions of TLS to be supported, TLS v1.3 will be made a minimum.
+
+#### Authorization
+All verification should be done on API level and API treats the client's common name (CN) in the certificate as the user's identity.
+Client not able interact with a job not created by them in any way (Stream, query Status, Stop for another user's job is not allowed).
+
+API not cache any user authorization data and such data should be provided for every API call.
+
+#### Proposed Protobuf
+Following [protobuf](https://protobuf.dev/programming-guides/proto3/) proposed for the client/server:
+
+```protobuf
+  syntax="proto3";
+
+  option go_package = "<package_name>/proto";
+
+  package proto;
+
+  service JobWorker {
+    rpc Start(JobCreateRequest) returns (JobResponse) {}
+    rpc Status(JobRequest) returns (JobStatusResponse) {}
+    rpc Stream(JobRequest) returns (stream OutputResponse) {}
+    rpc Stop(JobRequest) returns (JobStatusResponse) {}
+  }
+
+  // requests
+  message JobCreateRequest {
+    double  cpu = 1;
+    int64   memBytes = 2;
+    int64   ioBytesPerSecond = 3;
+    int64   command = 4;
+    repeated string args = 5;
+  }
+
+  message JobRequest {
+    string  uuid = 1;
+  }
+
+  // responses
+  message JobResponse {
+    string  uuid = 1;
+  }
+
+  message JobStatusResponse {
+    string  status = 1;
+    int32   exitCode = 2;
+    string  exitReason = 3;
+  } 
+
+  message OutputResponse {
+    bytes   content = 1;
```

Comment:
is the content from `stdout`?

Reply:
@greedy52 Yes, but because type names `**Output**Response' and I don't want call fields as _output_. 
Maybe it in future it can be use not just stdout, but out type of output. That's why I've choose **content** as field name.

But I'm open for any ideas for the field name :)

Reply:
And added little more details what kind output server stream to the client.

<img width="818" alt="Screenshot 2024-08-26 at 11 55 02 AM" src="https://github.com/user-attachments/assets/1b6c5ae1-2168-4f56-b2d9-6c443487a679">